lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb

```diff
@@ -0,0 +1,53 @@
+module RSpec
+  module Mocks
+    # Support for `exclude_stubbed_classes_from_subclasses` configuration.
+    #
+    # @private
+    class ExcludeStubbedClassesFromSubclasses
+      def self.enable!
+        return unless RUBY_VERSION >= "3.1"
+        return if Class.respond_to?(:subclasses_with_rspec_mocks)
+
+        require 'weakref'
+
+        mod_something = Module.new do
+          def excluded_subclasses
+            @excluded_subclasses ||= []
+            @excluded_subclasses.select(&:weakref_alive?).map do |ref|
+              begin
+                ref.__getobj__
+              rescue ::WeakRef::RefError
+                nil
+              end
+            end.compact
+          end
+
+          def exclude_subclass(constant)
+            @excluded_subclasses ||= []
+            @excluded_subclasses << ::WeakRef.new(constant)
+          end
+        end
+        RSpec::Mocks.extend(mod_something)
```

Comment:
On another thought, can we use this class to keep the list instead of RSpec::Mocks?

Reply:
I agree and changed for that.